unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/MyAccountHUD/BlockedListController.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.Social.Friends;
+using DCL.Tasks;
+using SocialFeaturesAnalytics;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace DCL.MyAccount
+{
+    public class BlockedListController
+    {
+        private readonly IBlockedListComponentView view;
+        private readonly DataStore dataStore;
+        private readonly IUserProfileBridge userProfileBridge;
+
+        private readonly IBlockedListApiBridge blockedListApiBridge;
+        private readonly ISocialAnalytics socialAnalytics;
+        private readonly IFriendsController friendsController;
+
+        private CancellationTokenSource lifeTimeCancellationToken;
+        private readonly CancellationTokenSource showBlockedUsersCancellationToken = new ();
+
+        private UserProfile ownUserProfile => userProfileBridge.GetOwn();
+
+        public BlockedListController(
+            IBlockedListComponentView view,
+            DataStore dataStore,
+            IUserProfileBridge userProfileBridge,
+            IBlockedListApiBridge blockedListApiBridge,
+            ISocialAnalytics socialAnalytics,
+            IFriendsController friendsController)
+        {
+            this.view = view;
+            this.dataStore = dataStore;
+            this.userProfileBridge = userProfileBridge;
+            this.blockedListApiBridge = blockedListApiBridge;
+            this.socialAnalytics = socialAnalytics;
+            this.friendsController = friendsController;
+
+            dataStore.myAccount.isMyAccountSectionVisible.OnChange += OnMyAccountSectionVisibleChanged;
+            dataStore.myAccount.openSection.OnChange += OnMyAccountSectionTabChanged;
+
+            ownUserProfile.OnUpdate += OnOwnUserProfileUpdated;
+            view.OnUnblockUser += UnblockUser;
+        }
+
+        private void OnMyAccountSectionVisibleChanged(bool isVisible, bool _)
+        {
+            if (isVisible)
+                OpenSection();
+            else
+                CloseSection();
+        }
+
+        private void OpenSection()
+        {
+            lifeTimeCancellationToken = lifeTimeCancellationToken.SafeRestart();
```

Comment:
Where are you using this `lifeTimeCancellationToken`? II'm only seeing its restart and disposing, but not the use of it in any async method.